scripts/git/check_submodule_init.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import sys
+
+
+def run():
+    # Checking the "git status"
+    git_check = os.popen("git status")
+    git_status = git_check.readline()
+
+    # no-op if not a git repository

```

Comment:
It helps to return early for checks like this, instead of nesting more deeply:

```python
if condition:
  return

...
```

vs.

```python
if not condition:
  ...
```

https://medium.com/swlh/return-early-pattern-3d18a41bba8 some reading material for background 🙂

Reply:
Thanks @ScottTodd , looking into this and will make the changes. Btw what can I do with the yapf linting?

Reply:
https://github.com/google/yapf

```
pip install yapf
yapf ./path/to/file.py
```
Should apply the formatting that lint step wants.

(On mobile and ending for the day now, can help more tomorrow)

Thanks for your work on this!

Reply:
Thanks again @ScottTodd , I will definitely try to do the changes.

Reply:
Made the changes @ScottTodd